NitroxServer/Communication/Packets/Processors/FootstepPacketProcessor.cs

```diff
@@ -0,0 +1,53 @@
+using NitroxModel.DataStructures.Unity;
+using NitroxModel.GameLogic.FMOD;
+using NitroxModel.Packets;
+using NitroxServer.Communication.Packets.Processors.Abstract;
+using NitroxServer.GameLogic;
+
+namespace NitroxServer.Communication.Packets.Processors;
+public class FootstepPacketProcessor : AuthenticatedPacketProcessor<FootstepPacket>
+{
+    private readonly float footstepAudioRange; // To modify this value, modify the last value in the SoundWhitelist_Subnautica.csv file
+    private readonly PlayerManager playerManager;
+
+    public FootstepPacketProcessor(PlayerManager playerManager, FMODWhitelist whitelist)
+    {
+        this.playerManager = playerManager;
+        whitelist.TryGetSoundData("event:/player/footstep_precursor_base", out SoundData soundData);
+        footstepAudioRange = soundData.Radius;
+    }
+
+    public override void Process(FootstepPacket footstepPacket, Player sendingPlayer)
+    {
+        var players = playerManager.GetAllPlayers();
+        foreach (Player player in players)
+        {
+            if (sendingPlayer.SubRootId.HasValue)
+            {
+                if (player.SubRootId.HasValue)
+                {
+                    // If both players have id's, check if they are the same
+                    if (NitroxVector3.Distance(player.Position, sendingPlayer.Position) <= footstepAudioRange && player != sendingPlayer && player.SubRootId.Value == sendingPlayer.SubRootId.Value)
```

Comment:
> Line is too long imo

Still not fixed or commented why keeping that way.


> These two ifs could be merged

> think I did that

With merging I mean that two nested ifs with no else can be merged into one statement, meaning you chain both conditions together with an `&&`

Reply:
Think I got this done actually this time but will keep conversation open until you mark it as resolved